src/main/java/tech/stackable/hadoop/StackableGroupMapper.java

```diff
@@ -0,0 +1,142 @@
+package tech.stackable.hadoop;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.GroupMappingServiceProvider;
+import org.apache.hadoop.util.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+public class StackableGroupMapper implements GroupMappingServiceProvider {
+    public static final String OPA_MAPPING_URL_PROP = "hadoop.security.group.mapping.opa.url";
+    public static final String OPA_MAPPING_GROUP_NAME_PROP = "hadoop.security.group.mapping.opa.list.name";
+    // response base field: see https://www.openpolicyagent.org/docs/latest/rest-api/#response-message
+    public static final String OPA_RESULT_FIELD = "result";
+    public final String mappingGroupName;
+    private final Logger LOG = LoggerFactory.getLogger(StackableGroupMapper.class);
+    private final HttpClient httpClient = HttpClient.newHttpClient();
+    private final ObjectMapper json;
+    private URI opaUri = null;
+
+    public StackableGroupMapper() {
+        Configuration configuration = new Configuration();
+
+        String opaMappingUrl = configuration.get(OPA_MAPPING_URL_PROP);
+        if (opaMappingUrl == null) {
+            throw new RuntimeException("Config \"" + OPA_MAPPING_URL_PROP + "\" missing");
+        }
+
+        try {
+            this.opaUri = URI.create(opaMappingUrl);
+        } catch (Exception e) {
+            throw new OpaException.UriInvalid(opaUri, e);
+        }
+
+        this.mappingGroupName = configuration.get(OPA_MAPPING_GROUP_NAME_PROP);
+        if (mappingGroupName == null) {
+            throw new RuntimeException("Config \"" + OPA_MAPPING_GROUP_NAME_PROP + "\" missing");
+        }
+
+        this.json = new ObjectMapper()
```

Comment:
Just FYI that I need specific `ObjectMapper` settings and will overwrite it with the followup PR :see_no_evil:

Reply:
In the end I went with a simple HashMap.class as the rego rule output should mean the groups is a named top-level key so I don't think different settings will matter.